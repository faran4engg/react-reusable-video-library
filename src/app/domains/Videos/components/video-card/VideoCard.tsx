import { FC } from 'react';
import { useHistory } from 'react-router-dom';
import VideoPlayer from './../video-player/VideoPlayer';
import VideoDetails from '../video-details/VideoDetails';
import { VideoCardProps } from './types';

const VideoCard: FC<VideoCardProps> = ({
  video,
  id,
  showMoreDetails = true,
  disbaleVideoClick,
  isLightMode = true,
  showWatchBtn,
}) => {
  const history = useHistory();
  const videoViewUrl = `/watch/${id}`;

  const watchVideo = () => {
    history.push(videoViewUrl);
  };
  return (
    <div className="overflow-hidden bg-transparent rounded-lg shadow-lg cursor-pointer">
      <VideoPlayer
        url={video.url}
        disbaleVideoClick={disbaleVideoClick}
        isLightMode={isLightMode}
        watchVideo={watchVideo}
      />
      <VideoDetails
        details={video}
        watchVideo={watchVideo}
        showMoreDetails={showMoreDetails}
        showWatchBtn={showWatchBtn}
        videoViewUrl={videoViewUrl}
      />
    </div>
  );
};

export default VideoCard;
