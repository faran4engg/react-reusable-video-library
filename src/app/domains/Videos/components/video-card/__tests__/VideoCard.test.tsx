import React from 'react';
import { act, create } from 'react-test-renderer';
import { createRenderer } from 'react-test-renderer/shallow';

import VideoCard from '../VideoCard';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const renderer = createRenderer();

describe('<VideoCard />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(
      <VideoCard title="Video Title" url="yt.com/abc" id="id" showWatchBtn />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should redirect to watch page', () => {
    const videoId = 'videoId';
    const renderComponent = () =>
      create(
        <VideoCard
          title="Video Title"
          url="yt.com/abc"
          id={videoId}
          showWatchBtn
        />,
      );
    const component = renderComponent();
    const root = component.root;
    const videoCard = root.findAllByType('div')[0];

    act(() => {
      videoCard.props.onClick();
    });

    expect(mockHistoryPush).toHaveBeenCalledTimes(1);
    expect(mockHistoryPush).toHaveBeenCalledWith(`/watch/${videoId}`);
  });
});
