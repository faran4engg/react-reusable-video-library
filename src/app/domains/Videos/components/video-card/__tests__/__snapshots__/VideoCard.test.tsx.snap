// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoCard /> should render and match the snapshot 1`] = `
<div
  className="overflow-hidden bg-transparent rounded-lg shadow-lg cursor-pointer"
  onClick={[Function]}
>
  <VideoPlayer
    isLightMode={true}
    url="yt.com/abc"
  />
  <VideoTitle
    showWatchBtn={true}
    title="Video Title"
    videoViewUrl="/watch/id"
  />
</div>
`;
